Sources/Core/Src/my_app/app_processing/app_processing.c

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file data_processing_api.c
+ * @author Nicolás Almaraz
+ * @brief This API resolves the data processing
+ */
+
+//======================================
+// Dependencies
+//======================================
+#include <app_inject_simulator/inject_simulator_api.h>
+#include <app_processing/app_processing.h>
+#include <app_processing/current_sensor_api/current_sensor_api.h>
+#include <app_processing/display_api/display_api.h>
+#include <app_processing/signal_analyzer_api/signal_analyzer_api.h>
+#include <stdbool.h>
+
+#include "main.h"
+#include "my_app.h"
+
+#include "common_apis/cycle_detector_api/cycle_detector_api.h"
+#include "common_apis/timer_api/timer_api.h"
+
+#include "signal_analyzer_api/signal_analyzer_api.h"
+
+//======================================
+// Private Defines
+//======================================
+
+//======================================
+// Private Data Structures and Types
+//======================================
+typedef enum {
+	STATE_RESET,
+	STATE_CALIBRATING,
+	STATE_STAND_BY,
+	STATE_SAMPLING,
+	STATE_PROCESSING,
+	STATE_WAITING_SETTING_TIME,
+}app_state_t;
+
+//======================================
+// Private Variables
+//======================================
+static app_state_t g_state = STATE_RESET;
```

Comment:
WHAT ?!?!
Si es estática, por qué la llamás `g_` ? Si vas a usar una nomenclatura (horrible, pero hay peores como la hungarian notation), al menos usala como es. Stanley Kubrick hubo uno solo, nadie va a aprender un idioma para entender tu código.

Reply:
Eso, estoy usando mal el coding style.
Tengo un error incorporado que dentro de mí cabeza las variables pueden ser "globales" o "globales privadas". Entonces a las dos le pongo g_. Evidentemente es un invento mío que está mal.

Lo voy a corregir, sino nadie me va a entender. Mejor usar un coding style estándar.

Reply:
Te puedo recomendar una psicóloga.
A ver, intentá explicarme cómo puede llegar a existir una variable "global privada" sin recurrir a security through obscurity.
Y esas variables, están entre nosotros ?

Reply:
Sergio, me quedó la duda.
En los ejemplos del libro cuando hace este tipo de comentarios:

//=====[Declaration and initialization of public global variables]=============

//=====[Declaration and initialization of private global variables]============

¿Puede ser que en realidad lo correcto sea poner "global variables" y "private variables"?

Reply:
In my defense (o defence, según el lado del charco), yo no tuve nada que ver con eso. Habría que preguntarle a los autores @alutenberg . Para mí, el "global" está de más.
Desconozco C++ y no sé si con las clases aparece alguna cosa rara. En Kernighan & Ritchie C, se definen:
- Variable types
_internal_ --> definidas dentro de una función ("privadas" (de la función), "locales")
_external_ --> definidas fuera de una función (*)

- Storage classes:
`static` --> se alojan en una posición fija de memoria 
si no se indica, son _automatic_ porque sólo existen mientras se ejecuta la función (se alojan en el stack)

_internal_ variables:
    a secas, se alojan en el stack ("auto")
    `static`, se alojan en una posición fija de memoria

_external_ variables: siempre se alojan en una posición fija de memoria, porque deben ser accesibles desde varias funciones o incluso módulos
    a secas, accesibles por quien las declara como `extern` ("globales", "públicas")
    `static`, accesibles sólo dentro del módulo en el que se las declara. ("privadas" (del módulo))

Los nombres entre comillas no corresponden a terminología de C.
`static` y `extern` se aplican de modo similar a funciones, aunque este último no es necesario (en una variable sí, porque sino creás almacenamiento de nuevo y colisionan)

Reply:
Creo entender que están aplicando el "globales" dentro del módulo, lo cual a mi criterio es incorrecto. O sea, llaman "globales" a las variables external, y luego las external static las llaman privadas y las external a secas las llaman públicas.

Las variables accesibles desde el exterior del módulo (que muchos consideran mala práctica y prefieren que se las encapsule y acceda mediante funciones, a la métodos de OO, deben declararse en el header (.h), para ser visibles por quien las llama. Esto genera un problema de multiple allocation que se cura con 2 curros. El más usado hoy, dado que ya los compiladores lo toleran (antes no...) es declararla `extern` en el header y definirla en el .c

Reply:
Entendido,
Entonces hablando en C:
Una variable puede ser global (= pública) ó privada. Puede ser una o la otra, no las dos en simultáneo.

Para ahorrar problemas de nomenclatura, saqué todo lo que antiguamente era:
```
static uint32_t g_variable_privada;
static void nombre_modulo_funcion_privada();
void nombre_modulo_funcion_publica();
``` 

Y ahora es:
```
static uint32_t m_variable_privada; //vi tarde la sugerencia de usar s_
static void m_funcion_privada();
void nombre_modulo_funcion_publica();
```

Las únicas variables globales (con extern) que utilizo son los handlers de los periféricos (htim, hdac, hadc, etc.) que genera el IDE en main.c. Sacando esas no debería haber quedado ninguna g_.

Gracias!

Reply:
> Una variable puede ser global (= pública) ó privada. Puede ser una o la otra, no las dos en simultáneo.

Una empresa puede ser una "sociedad mixta", pero las variables son públicas o privadas, que yo sepa no puede haber "un porcentaje de una variable" que se vea desde afuera...

Reply:
De las globales de la HAL no podemos zafar.
Yo uso globales, pocas pero uso.